src/models/evaluation/threshold_evaluator.py

```
@@ -364,7 +364,7 @@ def optimize_prediction_threshold(
                 f"Predictions range: [{test_predictions.min():.4f}, {test_predictions.max():.4f}]"
             )
         except Exception as e:
-            logger.error(f"❌ Failed to get predictions: {e}")
+            logger.error(f"Failed to get predictions: {e}")
```

Ruff (0.14.4)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)